2.13/mocks/mockOSB.js

```diff
@@ -187,11 +200,32 @@ app.delete('/v2/service_instances/:instance_id', function (req, res) {
 
   if (!req.query.service_id || !req.query.plan_id) {
     res.sendStatus(400)
+    return
+  }
+  
+  var found = findWhichContains(serviceInstances, 'instance_id', req.params.instance_id)
+  if (!found) {
+    res.status(410).send({})
+    return
   } else {
-    res.status(202).send({
-      'operation': 'task_10'
-    })
+    for (var i in serviceInstances) {
+      if (serviceInstances[i] === found) {
+        delete serviceInstances[i]
+      }
+    }
+    // Deprovision operation will also remove all binding info related to the instance

```

Comment:
I think the behavior below is not really for "all" bindings. It depends on the current mock config. The behavior is okay as a mock. But the comment should not be misleading.